packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -573,24 +573,22 @@ class AndroidDevice extends Device {
         ...<String>['--ez', 'trace-systrace', 'true'],
       if (debuggingOptions.dumpSkpOnShaderCompilation)
         ...<String>['--ez', 'dump-skp-on-shader-compilation', 'true'],
-      if (debuggingOptions.debuggingEnabled)
-        ...<String>[
-          if (debuggingOptions.buildInfo.isDebug)
-            ...<String>[
-              ...<String>['--ez', 'enable-checked-mode', 'true'],
-              ...<String>['--ez', 'verify-entry-points', 'true'],
-            ],
-          if (debuggingOptions.startPaused)
-            ...<String>['--ez', 'start-paused', 'true'],
-          if (debuggingOptions.disableServiceAuthCodes)
-            ...<String>['--ez', 'disable-service-auth-codes', 'true'],
-          if (debuggingOptions.dartFlags.isNotEmpty)
-            ...<String>['--es', 'dart-flags', debuggingOptions.dartFlags],
-          if (debuggingOptions.useTestFonts)
-            ...<String>['--ez', 'use-test-fonts', 'true'],
-          if (debuggingOptions.verboseSystemLogs)
-            ...<String>['--ez', 'verbose-logging', 'true'],
+      if (debuggingOptions.debuggingEnabled) ...<String>[
```

Comment:
This looks strange to me. Why is the if sometimes multi-line and sometimes not?

Reply:
From a formatting point of view putting `...<T>[` on the same line as `if (a)` is mainly depending on the presence of a trailing comma in the list.

```dart
// without trailing comma (v1)
if (route != null)
  ...<String>['--es', 'route', route],

// without trailing comma (v2)
if (route != null) ...<String>['--es', 'route', route],

// with trailing comma
if (route != null) ...<String>[
  '--es',
  'route',
  route,
],
```

The question to use multi-line/trailing-comma or not is more a question that depends on readability and understandability and the answer is in the responsibility of the code writer.

FWIW here's how `dartfmt` (with a line length at 120) format this code section:

```dart
    cmd = <String>[
      'shell', 'am', 'start',
      '-a', 'android.intent.action.RUN',
      '-f', '0x20000000', // FLAG_ACTIVITY_SINGLE_TOP
      '--ez', 'enable-background-compilation', 'true',
      '--ez', 'enable-dart-profiling', 'true',
      if (traceStartup) ...<String>['--ez', 'trace-startup', 'true'],
      if (route != null) ...<String>['--es', 'route', route],
      if (debuggingOptions.enableSoftwareRendering) ...<String>['--ez', 'enable-software-rendering', 'true'],
      if (debuggingOptions.skiaDeterministicRendering) ...<String>['--ez', 'skia-deterministic-rendering', 'true'],
      if (debuggingOptions.traceSkia) ...<String>['--ez', 'trace-skia', 'true'],
      if (debuggingOptions.traceSystrace) ...<String>['--ez', 'trace-systrace', 'true'],
      if (debuggingOptions.dumpSkpOnShaderCompilation) ...<String>['--ez', 'dump-skp-on-shader-compilation', 'true'],
      if (debuggingOptions.debuggingEnabled) ...<String>[
        if (debuggingOptions.buildInfo.isDebug) ...<String>[
          ...<String>['--ez', 'enable-checked-mode', 'true'],
          ...<String>['--ez', 'verify-entry-points', 'true'],
        ],
        if (debuggingOptions.startPaused) ...<String>['--ez', 'start-paused', 'true'],
        if (debuggingOptions.disableServiceAuthCodes) ...<String>['--ez', 'disable-service-auth-codes', 'true'],
        if (debuggingOptions.dartFlags.isNotEmpty) ...<String>['--es', 'dart-flags', debuggingOptions.dartFlags],
        if (debuggingOptions.useTestFonts) ...<String>['--ez', 'use-test-fonts', 'true'],
        if (debuggingOptions.verboseSystemLogs) ...<String>['--ez', 'verbose-logging', 'true'],
      ],
      apk.launchActivity,
    ];
```

You can see a similar formatting (except if one-liner) with `if` sometimes multi-line and sometimes not.

Do you prefer this output with `if` one-liner ?